### PR TITLE
UUIDs

### LINKED ISSUE #17: Use UUIDs for external reference
@phantompatrol suggested using [UUID](http://ruby-doc.org/stdlib-2.2.1/libdoc/securerandom/rdoc/SecureRandom.html#method-c-uuid)s to reference objects.

Going to pull from [jobapps / FriendlyId](https://github.com/Anbranin/jobapps/commit/702ec5f7608a825e4bd4a0bf3eb41a3366ae5636#diff-0e57bd675c6b231c68f7e02738aa7e1bR2) for this one.

Will need to update documentation, tests, controllers, and model files, so fairly large-scale enhancement.

### PR DESCRIPTION
**Pull request by akaplo**
Closes #17.  Adds support for generating and using (pretty much all abstracted away from the user of the API) universally unique identifiers (UUIDs) in place of standard, small int-valued IDs for all database entries.

Includes an updated README and specs.

### CODE REVIEW

**Comment by dfaulken** [comment]
Still not sure why the CodeClimate integration isn't running - I'll run it manually on my local machine.

**Comment by dfaulken** [comment]
Looks like it's just RuboCop failures - `rubocop -a` should take care of those when you get a chance.

File: pbcopy [hunk 1]
```diff
@@ -0,0 +1 @@
+"6076193a-a17b-4eae-a253-1b630485b822"
```
Comment (dfaulken) [inline pbcopy]: Whoops, let's delete this.

I meant to say `rails runner 'p SecureRandom.uuid' | pbcopy`, not `>`.

**Comment by dfaulken** [comment]
:warning: merge conflicts

**Comment by akaplo** [comment]
:zap @dfaulken should be good to go!

**Pull request merged**